validator/client/src/main/java/tech/pegasys/teku/validator/client/BeaconProposerPreparer.java

```diff
@@ -35,13 +35,15 @@
 import tech.pegasys.teku.validator.client.ProposerConfig.Config;
 import tech.pegasys.teku.validator.client.proposerconfig.ProposerConfigProvider;
 
-public class BeaconProposerPreparer implements ValidatorTimingChannel, FeeRecipientProvider {
+public class BeaconProposerPreparer
+    implements ValidatorTimingChannel, FeeRecipientProvider, GasLimitProvider {
```

Comment:
probably already clear, but i think `FeeRecipeintProvider` and `GasLimitProvider` is probably a level of granularity too low, we might just want something like `ValidatorRegistrationDetailsProvider` or something...

Reply:
It's all about finding a good generic name 😅

Reply:
I named it `ValidatorRegistrationPropertiesProvider`